From 4ae0eb4ccf04282bc38050df9f94b22f9017d5c9 Mon Sep 17 00:00:00 2001
From: Jon TURNEY <jon.turney@dronecode.org.uk>
Date: Tue, 14 Jul 2009 22:13:52 +0100
Subject: [PATCH 4/5] Cygwin/X: Fix system menu after window initialization re-ordering

Reorder so WS_SYSMENU style is set and WIN_WID_PROP propery has been set
before sending the WM_INIT_SYS_MENU message, so SetupSysMenu() will work
correctly now

Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
---
 hw/xwin/winmultiwindowwindow.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/hw/xwin/winmultiwindowwindow.c b/hw/xwin/winmultiwindowwindow.c
index 9bc527a..b05d310 100644
--- a/hw/xwin/winmultiwindowwindow.c
+++ b/hw/xwin/winmultiwindowwindow.c
@@ -584,7 +584,7 @@ winCreateWindowsWindow (WindowPtr pWin)
     }
  
   /* Change style back to popup, already placed... */
-  SetWindowLong (hWnd, GWL_STYLE, WS_POPUP | WS_CLIPCHILDREN | WS_CLIPSIBLINGS);
+  SetWindowLong (hWnd, GWL_STYLE, WS_POPUP | WS_SYSMENU | WS_CLIPCHILDREN | WS_CLIPSIBLINGS);
   SetWindowPos (hWnd, 0, 0, 0, 0, 0,
 		SWP_FRAMECHANGED | SWP_NOZORDER | SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE);
   /* Make sure it gets the proper system menu for a WS_POPUP, too */
@@ -592,14 +592,14 @@ winCreateWindowsWindow (WindowPtr pWin)
 
   pWinPriv->hWnd = hWnd;
 
-  /* Cause any .XWinrc menus to be added in main WNDPROC */
-  PostMessage (hWnd, WM_INIT_SYS_MENU, 0, 0);
-  
   SetProp (pWinPriv->hWnd, WIN_WID_PROP, (HANDLE) winGetWindowID(pWin));
 
   /* Flag that this Windows window handles its own activation */
   SetProp (pWinPriv->hWnd, WIN_NEEDMANAGE_PROP, (HANDLE) 0);
 
+  /* Cause any .XWinrc menus to be added in main WNDPROC */
+  PostMessage (hWnd, WM_INIT_SYS_MENU, 0, 0);
+
   /* Call engine-specific create window procedure */
   (*pScreenPriv->pwinFinishCreateWindowsWindow) (pWin);
 }
-- 
1.6.4.2

